packages/node-core/src/indexer/dictionary/v2/dictionaryV2.ts

```diff
@@ -0,0 +1,202 @@
+  endpoint: string,
+  axiosInstance?: AxiosInstance
+): Promise<DictionaryV2Metadata> {
+  if (!axiosInstance) {
+    axiosInstance = axios.create({
+      baseURL: endpoint,
+    });
+  }
+
+  const requestData = {
+    jsonrpc: '2.0',
+    method: FAT_META_QUERY_METHOD,
+    id: 1,
+  };
+  try {
+    const response = await axiosInstance.post<{
+      result?: DictionaryV2Capabilities;
+      error?: {code: number; message: string};
+    }>(endpoint, requestData, {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+    });
+
+    if (response.status >= 400) {
+      throw new Error(`[${response.status}]: ${response.statusText}`);
+    }
+
+    if (response.data.error) {
+      throw new Error(response.data.error.message);
+    }
+
+    const {result} = response.data;
+
+    if (!result) {
+      throw new Error('Malformed response');
+    }
+
+    const metadata: DictionaryV2Metadata = {
+      start: result.availableBlocks[0].startHeight,
+      end: result.availableBlocks[0].endHeight,
+      genesisHash: result.genesisHash,
+      filters: result.filters,
+      supported: result.supportedResponses,
+    };
+    return metadata;
+  } catch (error) {
+    // Handle the error as needed
+    throw new Error(`Dictionary v2 get capability failed ${error}`);
+  }
+}
+
+export abstract class DictionaryV2<
+  FB,
+  DS,
+  QE extends DictionaryV2QueryEntry = DictionaryV2QueryEntry
+> extends CoreDictionary<DS, FB, DictionaryV2Metadata, QE> {
+  protected dictionaryApi: AxiosInstance;
+
+  constructor(
+    readonly dictionaryEndpoint: string,
+    protected chainId: string,
+    protected readonly nodeConfig: NodeConfig,
+    protected readonly eventEmitter: EventEmitter2
+  ) {
+    super(dictionaryEndpoint, chainId, nodeConfig, eventEmitter);
+    this.dictionaryApi = axios.create({
+      baseURL: dictionaryEndpoint,
+    });
+  }
+
+  static async isDictionaryV2(endpoint: string, timeoutSec = 1): Promise<boolean> {
+    let resp: DictionaryV2Metadata;
+    const timeoutMsg = 'Inspect dictionary version timeout';
+    try {
+      resp = await timeout(subqlFilterBlocksCapabilities(endpoint), timeoutSec, timeoutMsg);
+      return resp.supported.includes('complete') || resp.supported.includes('basic');
+    } catch (e: any) {
+      // TODO does it make sense to log here?
+      return false;
+    }
+  }
+
+  protected abstract buildDictionaryQueryEntries(dataSources: DS[]): QE;
+
+  async init(): Promise<void> {
+    this._metadata = await subqlFilterBlocksCapabilities(this.dictionaryEndpoint);
+    this.setDictionaryStartHeight(this._metadata.start);
+  }
+
+  getQueryEndBlock(targetBlockHeight: number, apiFinalizedHeight: number): number {
+    return Math.min(targetBlockHeight, this.metadata.end);
+  }
+
+  abstract convertResponseBlocks<RFB>(
+    result: RawFatDictionaryResponseData<RFB>
+  ): DictionaryResponse<IBlock<FB>> | undefined;
+
+  protected async coreGetData<RFB>(
```

Comment:
If there's no need to have a super version of this function then this can be exposed
```suggestion
 async getData<RFB>(
```